### DISCUSSION TITLE
502 Bad Gateway error after updating API KEY

### DISCUSSION THREAD

**Discussion by diflorin** [Budibase/budibase#11303] [Help]
Hi, I've updated the budibase api key as the colleague that has installed it is not in the team anymore and now we access it at all, all we get it a 502 Bad Gateway error. We have found the old api key. Can anyone please help us or even give us a hint on how to restore the instalation we had? Regards,

**Comment by ConorWebb96** [comment]
Hey @diflorin,

I have a few questions on the above.

- What can't you access anymore? The API key is normally used for using the public API. However this is normally outside of Budibase and wouldn't affect your system.
- Are you self-hosted or cloud?
- What version of Budibase are you running?
- What are you using the Budibase API key for?
- Can you send through a screenshot of your console were more details on the error are likely to be provided?
- Can you also send through a screenshot of your network requests and click on the error so we can see additional information on this?

**Comment by diflorin** [comment]
Hi Connor,

We're not 100% sure that the cause is the API KEY. But it was the last
thing we've changed on Monday and we thought that with the new API KEY some
elements were updated resulting in this incompatibility.

We can't access at all the login page - which is hosted in our subdomain
apps.datainnovation.io
Self hosted.
2.8.10 // it was updated automatically 3 days ago
I've updated the API Key because I was exploring the possibility to upgrade
from FREE to BUSINESS

CONTAINER ID        IMAGE
 COMMAND                  CREATED             STATUS                  PORTS
                                         NAMES
b8cf2a8f7a1c        budibase.docker.scarf.sh/budibase/apps:latest
 "docker-entrypoint.s…"   46 hours ago        Up 11 hours
4001/tcp                                       bbapps
854dfa953b75        budibase/proxy:latest
 "/docker-entrypoint.…"   3 days ago          Up 11 hours
80/tcp, 0.0.0.0:10000->10000/tcp               bbproxy
d854e553eb5d        budibase.docker.scarf.sh/budibase/worker:latest
 "docker-entrypoint.s…"   3 days ago          Up 11 hours
4001/tcp                                       bbworker
bb3ba53d7193        0256c63af7db
"docker-entrypoint.s…"   6 months ago        Up 11 hours
6379/tcp                                       mybudibase-redis-service-1
f54c0f50ea8d        minio/minio
 "/usr/bin/docker-ent…"   6 months ago        Up 11 hours (healthy)
9000/tcp                                       mybudibase-minio-service-1
50012a4f98f8        containrrr/watchtower
 "/watchtower --debug…"   6 months ago        Up 11 hours
8080/tcp
mybudibase-watchtower-service-1
a36d8877e15f        ibmcom/couchdb3
 "/docker-entrypoint.…"   6 months ago        Up 11 hours
4369/tcp, 5984/tcp, 9100/tcp                   mybudibase-couchdb-service-1
da302f5b2ae9        curlimages/curl
 "/entrypoint.sh sh -…"   6 months ago        Exited (2) 11 hours ago
                                         mybudibase-couch-init-1

My colleagues @Iván Martinez ***@***.***> & @Pau Figueras Pavon
***@***.***> know a bit more

Could you please help us? It

Regards,





Florin Armasu
CEO & Founder, Data Innovation
Meeting: www.calendly.com/diflorin
<http://www.calendly.com/diflorin?email=diflorin>
Call or WhatsApp: +34610199993 <https://wa.me/0034610199993>

**Reply by ConorWebb96** [reply]
Hey @diflorin,

When you said `API key` do you mean `INTERNAL_API_KEY `?

- Can you provide a list of steps you followed up to this point?
- Could you also provide network logs from your browser of the screen were you are receiving the error. (You can do this by right clicking the browser and clicking inspect element, then going to the network tab in this side window.)
- How did you update the API key?
- Did you update this within your `.env` file?

**Comment by diflorin** [comment]
Yes, thank you very much for following up.

The list of steps we did to try to restore it? I did update the internal
API Key directly in the interface -> now since I cannot enter I can't pass
you a screenshot. @Iván Martinez ***@***.***> please comment on
this.

The .har file is attached.

The original installation for Budibase was first configured by
***@***.*** or  ***@***.*** and since he's not working
with us anymore, but we need the application functioning I've tried to
activate an API KEY with my email.

And no, I don't believe we tried to update the .env file, but I can do it
right now.

Regards,



Florin Armasu
CEO & Founder, Data Innovation
Meeting: www.calendly.com/diflorin
<http://www.calendly.com/diflorin?email=diflorin>
Call or WhatsApp: +34610199993 <https://wa.me/0034610199993>
